pkg/queue/concurrency_state.go

```diff
@@ -103,52 +134,90 @@ func ConcurrencyStateHandler(logger *zap.SugaredLogger, h http.Handler, pause, r
 	}
 }
 
+// FreezePodRetry handle the failed request when call Resume/Pause
+func FreezePodRetry(logger *zap.SugaredLogger, ch chan RetryElement, retryDoneChannel chan struct{}, token *Token, pause, resume func(string, *Token) (int8, error)) {
+	for {
+		elementNow := <-ch
+		elementNow.timesNow += 1
+		if elementNow.timesNow > FreezeMaxRetryTimes {
+			// TODO: relaunch a new pod
+			panic("Relaunch a pod")
+		}
+		switch elementNow.op {
+		case "pause":
+			errCode, err := pause(elementNow.endpoint, token)
+			if errCode != responseStatusConflictError && errCode != noError {
+				logger.Info("Error handling pause request: %v", err)
+				ch <- elementNow
+				time.Sleep(100 * time.Millisecond)
+			} else {
+				retryDoneChannel <- struct{}{}
+			}
+		case "resume":
+			errCode, err := resume(elementNow.endpoint, token)
+			if errCode != responseStatusConflictError && errCode != noError {
+				logger.Info("Error handling resume request: %v", err)
+				ch <- elementNow
+				time.Sleep(100 * time.Millisecond)
+			} else {
+				retryDoneChannel <- struct{}{}
+			}
+		}
+	}
+}
+
 // Pause sends to an endpoint when request counts drop to zero
-func Pause(endpoint string, token *Token) error {
+func Pause(endpoint string, token *Token) (int8, error) {
```

Comment:
FYI I've deduped this a bit to only have a single function, rather than separate Pause / Resume functions ([link](https://github.com/knative/serving/pull/11802/commits/835feffa130b352401ed486e2f91d55cddf39250))